src/util/kaldi-table-inl.h

```diff
@@ -199,11 +227,15 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
     if (state_ == kLoadSucceeded) {
       holder_.Swap(other_holder);
       state_ = kLoadFailed;
+    } else if (state_ == kRangeExtracted) {
+      range_holder_.Swap(other_holder);
+      state_ = kLoadFailed;
```

Comment:
at this point I think the state should be set to kLoadSucceeded, indicating that we still have the base object (but no range).